lib/resources/mysql_session.rb

```diff
@@ -31,7 +32,16 @@ def query(q, db = '')
       escaped_query = q.gsub(/\\/, '\\\\').gsub(/"/, '\\"').gsub(/\$/, '\\$')
 
       # run the query
-      cmd = inspec.command("mysql -u#{@user} -p#{@pass} -h #{@host} #{db} -s -S #{@socket} -e \"#{escaped_query}\"")
+      command = "mysql -u#{@user} -p#{@pass}"
+      if !socket.nil?
+        command += " -S #{@socket}"
+      else
+        command += " -h #{@host}"
+      end
+      command += " --port #{@port}" unless @port.nil?
+      command += " #{db} -s -S #{@socket} -e \"#{escaped_query}\""
```

Comment:
Yep, the current version breaks for all queries without socket connections because of this